parse.go

```diff
@@ -273,10 +273,12 @@ func (c *Client) Do(req *http.Request, body, result interface{}) (*http.Response
 	}
 
 	res, err := c.RoundTrip(req)
+	if res.Body != nil {
```

Comment:
ah, i think i see the real issue which is that we need a `res.Body.Close()` after the `ioutil.ReadAll` on line 236 above. this change may panic because `res` itself may be nil.